server.go

```diff
@@ -639,6 +639,7 @@ func (s *Server) ReadTCPSocket() {
 			case <-s.shutdown:
 				// occurs when cleanly shutting down the server e.g. in tests; ignore errors
 				log.WithError(err).Info("Ignoring Accept error while shutting down")
+				return
```

Comment:
If we're returning here, should we have the `log.Fatal` in the default section, then? (That's the only path in which it gets executed).